src/tests/harness/BenchmarkDotNet.Extensions.Tests/UniqueValuesGeneratorTests.cs

```diff
@@ -72,5 +72,115 @@ public void GetNonDefaultValueReturnsNonDefaultValue()
             Assert.NotEqual(default(int), ValuesGenerator.GetNonDefaultValue<int>());
             Assert.NotEqual(default(string), ValuesGenerator.GetNonDefaultValue<string>());
         }
+
+        [Fact]
+        public void SupportsByte()
+        {
+            Supports<byte>();
+        }
+
+        [Fact]
+        public void SupportsChar()
+        {
+            Supports<char>();
+        }
+
+        [Fact]
+        public void SupportsShort()
+        {
+            Supports<short>();
+        }
+
+        [Fact]
+        public void SupportsUnsignedShort()
+        {
+            Supports<ushort>();
+        }
+
+        [Fact]
+        public void SupportsInteger()
+        {
+            Supports<int>();
+        }
+
+        [Fact]
+        public void SupportsUnsignedInteger()
+        {
+            Supports<uint>();
+        }
+
+        [Fact]
+        public void SupportsLong()
+        {
+            Supports<long>();
+        }
+ 
+        [Fact]
+        public void SupportsUnsignedLong()
+        {
+            Supports<ulong>();
+        }
+
+        [Fact]
+        public void SupportsFloat()
+        {
+            Supports<float>();
+        }
+
+        [Fact]
+        public void SupportsDouble()
+        {
+            Supports<double>();
+        }
+
+        [Fact]
+        public void SupportsBool()
+        {
+            SupportsArray<bool>();
+        }
+
+        [Fact]
+        public void SupportsDecimal()
+        {
+            Supports<decimal>();
+        }
+
+        [Fact]
+        public void SupportsString()
+        {
+            Supports<string>();
+        }
+
+        [Fact]
+        public void SupportsGuid()
+        {
+            Supports<Guid>();
+        }
+
+        private static void SupportsArray<T>()
+        {
+            var array = ValuesGenerator.Array<T>(10);
+            Assert.NotNull(array);
+            Assert.Equal(10, array.Length);
+        }
+        private static void SupportsArrayOfUniques<T>()
+        {
+            var array = ValuesGenerator.ArrayOfUniqueValues<T>(5);
+            Assert.NotNull(array);
+            Assert.Equal(5, array.Length);
+        }
+
+        private static void SupportsNonDefaultValue<T>()
+        {
+            var value = ValuesGenerator.GetNonDefaultValue<T>();
+            Assert.NotEqual(default(T), value);
+        }
+
+        private static void Supports<T>()
+        {
+            SupportsArray<T>();
+            SupportsNonDefaultValue<T>();
+            SupportsArrayOfUniques<T>();
```

Comment:
could you please also add a method for testing `Dictionary` support?

https://github.com/dotnet/performance/blob/3e63b12c2cfe8d3f4f8b6083e49ab28b5111cedd/src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs#L100

Reply:
Since the dictionary has a `TKey` and a `TValue`, but only the type of the `TKey` matters to the value generation logic, do you have a preference for the type used for `TValue`? 

- `TValue` could be same type as `TVal`, and the value stored would be the key's value
- `TValue` could be `string` and be the  string representation of the key value generated, or a random `string` value
- `TValue` could just be `int` and be the `(int)` cast of the key value generated, or a random `int` value

For now, unless you express another choice, will do the first option.

Reply:
Done